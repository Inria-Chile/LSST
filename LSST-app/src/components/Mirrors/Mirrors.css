.mirrors-container {
    display: flex;
    flex-direction: column;
    margin: 5px;
    height: 100%;
}

.mirrors-container h2 {
    margin: 0px;
    margin-bottom: 5px;
    color: #fff;
}

.mirrors-container h4 {
    background: #3a4d61;
    font-weight: normal;
}

.mirrors-container .container {
    background: #3a4d61;
    width: 100%;
    margin: 0px;
    height: 100%;
}

.m1m3-heading-col {
    padding: 0px;
    padding-left: 5px;
}

.m2-heading-col {
    padding: 0px;
}

.mirror-row {
    margin-bottom: 50px;
}

.mirror-bottom-row {
    margin-bottom: 25px;
}