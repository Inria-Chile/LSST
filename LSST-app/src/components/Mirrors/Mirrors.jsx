import React, { Component } from 'react';
// import ReactDOM from 'react-dom';
import M1M3 from './M1M3/M1M3';
import ColorScaleLegend from './ColorScaleLegend/ColorScaleLegend';
import './Mirrors.css';
import * as d3 from 'd3';

class Mirrors extends Component {

    static viewName = 'mirrors';
    
    constructor(props) {
        super(props);
        this.state = {
        }
    }

    render() {
        let mirrorSize = 320;
        let mirrorMargin = 30;
        let colormap1 = d3.scaleSequential((t) => d3.hsl(360, 1.0-t*t*0.1, 0.12+t*t*0.58) + "");
        let colormap2 = d3.scaleSequential((t) => d3.hsl(270, 0.9+t*t*0.1, 0.7-t*t*0.58) + "");
        let colormap3 = d3.scaleSequential((t) => d3.hsl(140, 0.9+t*t*0.1, 0.7-t*t*0.58) + "");
        return (
            <div className="mirrors-container">
                <div>
                    <h2>
                        Mirrors
                    </h2>
                </div>
                <div className="container pull-left">
                    <div className="row">
                        <div className="col-6 m1m3-heading-col">
                            <h4>M1M3</h4>
                        </div>
                        <div className="col-6 m2-heading-col">
                            <h4>M2</h4>
                        </div>
                    </div>
                    <div className="row mirror-row">
                        <div className="col-12">
                            <h5>Temperature</h5>
                        </div>
                        <div className="col-5">
                            <M1M3 width={mirrorSize} height={mirrorSize} id="m3" scale={0.83} margin={mirrorMargin} colormap={colormap1}/>
                        </div>
                        <div className="col-5">
                            <M1M3 width={mirrorSize} height={mirrorSize} id="m4" scale={0.83} margin={mirrorMargin} colormap={colormap1}/>
                        </div>
                        <div className="col-2">
                            <ColorScaleLegend max={10} min={0} width={mirrorSize/3} height={mirrorSize} colormapID={"colormap1"} colormap={colormap1}/>
                        </div>
                    </div>
                    <div className="row mirror-row">
                        <div className="col-12">
                            <h5>Force (actuators)</h5>
                        </div>
                        <div className="col-5">
                            <M1M3 width={mirrorSize} height={mirrorSize} id="m5" scale={0.83} margin={mirrorMargin} colormap={colormap2}/>
                        </div>
                        <div className="col-5">
                            <M1M3 width={mirrorSize} height={mirrorSize} id="m6" scale={0.83} margin={mirrorMargin} colormap={colormap2}/>
                        </div>
                        <div className="col-2">
                            <ColorScaleLegend max={10} min={0} width={mirrorSize/3} height={mirrorSize} colormapID={"colormap2"} colormap={colormap2}/>
                        </div>
                    </div>
                    <div className="row mirror-row">
                        <div className="col-12">
                            <h5>Force (hard points)</h5>
                        </div>
                        <div className="col-5">
                            <M1M3 width={mirrorSize} height={mirrorSize} id="m7" scale={0.83} margin={mirrorMargin} colormap={colormap3}/>
                        </div>
                        <div className="col-5">
                            <M1M3 width={mirrorSize} height={mirrorSize} id="m8" scale={0.83} margin={mirrorMargin} colormap={colormap3}/>
                        </div>
                        <div className="col-2">
                            <ColorScaleLegend max={10} min={0} width={mirrorSize/3} height={mirrorSize} colormapID={"colormap3"} colormap={colormap3}/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default Mirrors;
