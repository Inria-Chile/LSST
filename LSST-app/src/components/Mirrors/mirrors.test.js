import React from 'react';
import Mirrors from './Mirrors';
import renderer from 'react-test-renderer';
import * as d3 from 'd3';


it('renders correctly', () => {
  const tree = renderer
    .create(<Mirrors/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

