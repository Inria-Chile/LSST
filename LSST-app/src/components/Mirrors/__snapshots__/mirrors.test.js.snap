// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mirrors-container"
>
  <div
    className="draggable-title "
  >
    <h2>
      Mirrors
    </h2>
    <div
      className="move-button"
    >
      <svg
        fill="currentColor"
        height={30}
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width={30}
      >
        <g>
          <path
            d="m29.5 13.8c-0.6-0.6-1.7-0.6-2.3 0s-0.7 1.7 0 2.4l2.1 2.1h-7.6v-7.6l2.1 2.1c0.3 0.4 0.8 0.5 1.2 0.5s0.9-0.1 1.2-0.5c0.6-0.6 0.6-1.7 0-2.3l-6.2-6.2-6.2 6.2c-0.6 0.6-0.6 1.7 0 2.3s1.7 0.7 2.4 0l2.1-2.1v7.6h-7.6l2.1-2.1c0.7-0.7 0.7-1.7 0-2.4s-1.7-0.6-2.3 0l-6.2 6.2 6.2 6.2c0.3 0.3 0.7 0.5 1.2 0.5s0.8-0.2 1.1-0.5c0.7-0.7 0.7-1.7 0-2.4l-2.1-2.1h7.6v7.6l-2.1-2.1c-0.7-0.7-1.7-0.7-2.4 0s-0.6 1.7 0 2.3l6.2 6.2 6.2-6.2c0.6-0.6 0.6-1.7 0-2.3s-1.7-0.7-2.4 0l-2.1 2.1v-7.6h7.6l-2.1 2.1c-0.7 0.7-0.7 1.7 0 2.4 0.3 0.3 0.7 0.5 1.1 0.5s0.9-0.2 1.2-0.5l6.2-6.2-6.2-6.2z"
          />
        </g>
      </svg>
    </div>
    
  </div>
  <div
    className="container pull-left"
  >
    <div
      className="row"
    >
      <div
        className="col-6 m1m3-heading-col"
      >
        <h4>
          M1M3
        </h4>
      </div>
      <div
        className="col-6 m2-heading-col"
      >
        <h4>
          M2
        </h4>
      </div>
    </div>
    <div
      className="row mirror-row"
    >
      <div
        className="col-12"
      >
        <h5>
          Temperature
        </h5>
      </div>
      <div
        className="col-5"
      >
        <div
          className="mirror-container"
        >
          <svg
            className="svg-container"
            height="250px"
            width="250px"
          >
            <circle
              className="circle-overlay"
              cx={30}
              cy={30}
              fill="#04070a"
              id="background-circle-m3"
              pointerEvents="all"
              r={0}
            />
            <g
              className="scatter"
              id="scatter-m3"
            />
            <circle
              cx={30}
              cy={30}
              fill="none"
              id="circle-overlay-m3"
              pointerEvents="all"
              r={0}
            />
          </svg>
        </div>
      </div>
      <div
        className="col-5"
      >
        <div
          className="mirror-container"
        >
          <svg
            className="svg-container"
            height="250px"
            width="250px"
          >
            <circle
              className="circle-overlay"
              cx={30}
              cy={30}
              fill="#04070a"
              id="background-circle-m4"
              pointerEvents="all"
              r={0}
            />
            <g
              className="scatter"
              id="scatter-m4"
            />
            <circle
              cx={30}
              cy={30}
              fill="none"
              id="circle-overlay-m4"
              pointerEvents="all"
              r={0}
            />
          </svg>
        </div>
      </div>
      <div
        className="col-2"
      >
        <div
          className="legend-container"
        >
          <svg
            height={250}
            id="legend-svg"
            width={83.33333333333333}
          >
            <g
              transform="translate(20,10)"
            >
              <defs>
                <linearGradient
                  id="colormap1"
                  spreadMethod="pad"
                  x1="0%"
                  x2="0%"
                  y1="100%"
                  y2="0%"
                >
                  <stop
                    offset="0%"
                    stopColor="rgb(61, 0, 0)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="11.11111111111111%"
                    stopColor="rgb(65, 0, 0)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="22.22222222222222%"
                    stopColor="rgb(76, 0, 0)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="33.33333333333333%"
                    stopColor="rgb(94, 1, 1)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="44.44444444444444%"
                    stopColor="rgb(118, 1, 1)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="55.55555555555556%"
                    stopColor="rgb(150, 2, 2)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="66.66666666666666%"
                    stopColor="rgb(188, 4, 4)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="77.77777777777779%"
                    stopColor="rgb(233, 7, 7)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="88.88888888888889%"
                    stopColor="rgb(247, 48, 48)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="100%"
                    stopColor="rgb(247, 110, 110)"
                    stopOpacity="1"
                  />
                </linearGradient>
              </defs>
              <rect
                className="colormap-rect"
                height={230}
                style={
                  Object {
                    "fill": "url(\\"#colormap1\\")",
                  }
                }
                transform="translate(16.666666666666664,0)"
                width={16.666666666666664}
                x1="0"
                y1="0"
              />
              <g
                className="legend axis"
                transform="translate(0, 0)"
              >
                <g
                  className="tick"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  transform="translate(0,230)"
                >
                  <line
                    x1={14}
                    x2={20}
                    y1="0"
                    y2="0"
                  />
                  <text
                    dy=".32em"
                    style={
                      Object {
                        "textAnchor": "end",
                      }
                    }
                    x="9"
                    y="0"
                  >
                    0
                  </text>
                </g>
                <g
                  className="tick"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  transform="translate(0,0)"
                >
                  <line
                    x1={14}
                    x2={20}
                    y1="0"
                    y2="0"
                  />
                  <text
                    dy=".32em"
                    style={
                      Object {
                        "textAnchor": "end",
                      }
                    }
                    x="9"
                    y="0"
                  >
                    10
                  </text>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="row mirror-row"
    >
      <div
        className="col-12"
      >
        <h5>
          Force (actuators)
        </h5>
      </div>
      <div
        className="col-5"
      >
        <div
          className="mirror-container"
        >
          <svg
            className="svg-container"
            height="250px"
            width="250px"
          >
            <circle
              className="circle-overlay"
              cx={30}
              cy={30}
              fill="#04070a"
              id="background-circle-m5"
              pointerEvents="all"
              r={0}
            />
            <g
              className="scatter"
              id="scatter-m5"
            />
            <circle
              cx={30}
              cy={30}
              fill="none"
              id="circle-overlay-m5"
              pointerEvents="all"
              r={0}
            />
          </svg>
        </div>
      </div>
      <div
        className="col-5"
      >
        <div
          className="mirror-container"
        >
          <svg
            className="svg-container"
            height="250px"
            width="250px"
          >
            <circle
              className="circle-overlay"
              cx={30}
              cy={30}
              fill="#04070a"
              id="background-circle-m6"
              pointerEvents="all"
              r={0}
            />
            <g
              className="scatter"
              id="scatter-m6"
            />
            <circle
              cx={30}
              cy={30}
              fill="none"
              id="circle-overlay-m6"
              pointerEvents="all"
              r={0}
            />
          </svg>
        </div>
      </div>
      <div
        className="col-2"
      >
        <div
          className="legend-container"
        >
          <svg
            height={250}
            id="legend-svg"
            width={83.33333333333333}
          >
            <g
              transform="translate(20,10)"
            >
              <defs>
                <linearGradient
                  id="colormap2"
                  spreadMethod="pad"
                  x1="0%"
                  x2="0%"
                  y1="100%"
                  y2="0%"
                >
                  <stop
                    offset="0%"
                    stopColor="rgb(179, 110, 247)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="11.11111111111111%"
                    stopColor="rgb(177, 106, 247)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="22.22222222222222%"
                    stopColor="rgb(171, 95, 247)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="33.33333333333333%"
                    stopColor="rgb(162, 77, 247)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="44.44444444444444%"
                    stopColor="rgb(149, 52, 247)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="55.55555555555556%"
                    stopColor="rgb(133, 19, 247)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="66.66666666666666%"
                    stopColor="rgb(113, 6, 219)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="77.77777777777779%"
                    stopColor="rgb(89, 4, 175)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="88.88888888888889%"
                    stopColor="rgb(62, 1, 122)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="100%"
                    stopColor="rgb(31, 0, 61)"
                    stopOpacity="1"
                  />
                </linearGradient>
              </defs>
              <rect
                className="colormap-rect"
                height={230}
                style={
                  Object {
                    "fill": "url(\\"#colormap2\\")",
                  }
                }
                transform="translate(16.666666666666664,0)"
                width={16.666666666666664}
                x1="0"
                y1="0"
              />
              <g
                className="legend axis"
                transform="translate(0, 0)"
              >
                <g
                  className="tick"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  transform="translate(0,230)"
                >
                  <line
                    x1={14}
                    x2={20}
                    y1="0"
                    y2="0"
                  />
                  <text
                    dy=".32em"
                    style={
                      Object {
                        "textAnchor": "end",
                      }
                    }
                    x="9"
                    y="0"
                  >
                    0
                  </text>
                </g>
                <g
                  className="tick"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  transform="translate(0,0)"
                >
                  <line
                    x1={14}
                    x2={20}
                    y1="0"
                    y2="0"
                  />
                  <text
                    dy=".32em"
                    style={
                      Object {
                        "textAnchor": "end",
                      }
                    }
                    x="9"
                    y="0"
                  >
                    10
                  </text>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="row mirror-row mirror-bottom-row"
    >
      <div
        className="col-12"
      >
        <h5>
          Force (hard points)
        </h5>
      </div>
      <div
        className="col-5"
      >
        <div
          className="mirror-container"
        >
          <svg
            className="svg-container"
            height="250px"
            width="250px"
          >
            <circle
              className="circle-overlay"
              cx={30}
              cy={30}
              fill="#04070a"
              id="background-circle-m7"
              pointerEvents="all"
              r={0}
            />
            <g
              className="scatter"
              id="scatter-m7"
            />
            <circle
              cx={30}
              cy={30}
              fill="none"
              id="circle-overlay-m7"
              pointerEvents="all"
              r={0}
            />
          </svg>
        </div>
      </div>
      <div
        className="col-5"
      >
        <div
          className="mirror-container"
        >
          <svg
            className="svg-container"
            height="250px"
            width="250px"
          >
            <circle
              className="circle-overlay"
              cx={30}
              cy={30}
              fill="#04070a"
              id="background-circle-m8"
              pointerEvents="all"
              r={0}
            />
            <g
              className="scatter"
              id="scatter-m8"
            />
            <circle
              cx={30}
              cy={30}
              fill="none"
              id="circle-overlay-m8"
              pointerEvents="all"
              r={0}
            />
          </svg>
        </div>
      </div>
      <div
        className="col-2"
      >
        <div
          className="legend-container"
        >
          <svg
            height={250}
            id="legend-svg"
            width={83.33333333333333}
          >
            <g
              transform="translate(20,10)"
            >
              <defs>
                <linearGradient
                  id="colormap3"
                  spreadMethod="pad"
                  x1="0%"
                  x2="0%"
                  y1="100%"
                  y2="0%"
                >
                  <stop
                    offset="0%"
                    stopColor="rgb(110, 247, 156)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="11.11111111111111%"
                    stopColor="rgb(106, 247, 153)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="22.22222222222222%"
                    stopColor="rgb(95, 247, 146)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="33.33333333333333%"
                    stopColor="rgb(77, 247, 134)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="44.44444444444444%"
                    stopColor="rgb(52, 247, 117)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="55.55555555555556%"
                    stopColor="rgb(19, 247, 95)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="66.66666666666666%"
                    stopColor="rgb(6, 219, 77)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="77.77777777777779%"
                    stopColor="rgb(4, 175, 61)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="88.88888888888889%"
                    stopColor="rgb(1, 122, 42)"
                    stopOpacity="1"
                  />
                  <stop
                    offset="100%"
                    stopColor="rgb(0, 61, 20)"
                    stopOpacity="1"
                  />
                </linearGradient>
              </defs>
              <rect
                className="colormap-rect"
                height={230}
                style={
                  Object {
                    "fill": "url(\\"#colormap3\\")",
                  }
                }
                transform="translate(16.666666666666664,0)"
                width={16.666666666666664}
                x1="0"
                y1="0"
              />
              <g
                className="legend axis"
                transform="translate(0, 0)"
              >
                <g
                  className="tick"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  transform="translate(0,230)"
                >
                  <line
                    x1={14}
                    x2={20}
                    y1="0"
                    y2="0"
                  />
                  <text
                    dy=".32em"
                    style={
                      Object {
                        "textAnchor": "end",
                      }
                    }
                    x="9"
                    y="0"
                  >
                    0
                  </text>
                </g>
                <g
                  className="tick"
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  transform="translate(0,0)"
                >
                  <line
                    x1={14}
                    x2={20}
                    y1="0"
                    y2="0"
                  />
                  <text
                    dy=".32em"
                    style={
                      Object {
                        "textAnchor": "end",
                      }
                    }
                    x="9"
                    y="0"
                  >
                    10
                  </text>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
